tests/unit/workflows/dotnet_clipackage/test_actions.py

```diff
@@ -42,6 +45,24 @@ def test_global_tool_update_failed(self):
         action = GlobalToolInstallAction(self.subprocess_dotnet)
         self.assertRaises(ActionFailedError, action.execute)
 
+    def test_global_tool_parallel(self):
+        executor = ThreadPoolExecutor()
+
+        actions = [
+            GlobalToolInstallAction(self.subprocess_dotnet),
+            GlobalToolInstallAction(self.subprocess_dotnet),
+            GlobalToolInstallAction(self.subprocess_dotnet),
+        ]
+
+        for action in actions:
+            executor.submit(action.execute)
+
+        executor.shutdown()
+
+        self.subprocess_dotnet.assert_has_calls(
```

Comment:
It seems we want to check that only 1 call to install was made.
You should maybe use this instead? 
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once_with

Reply:
Good catch, will update this